src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2515,14 +2517,15 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
                 try
                 {
                     if (SerializePeToStream(
-                        new EmitContext(moduleBeingBuilt, null, metadataDiagnostics),
+                        new EmitContext(moduleBeingBuilt, null, metadataDiagnostics, isRefAssembly: metadataOnly && !includePrivateMembers),
```

Comment:
>isRefAssembly: metadataOnly && !includePrivateMembers [](start = 85, length = 53)

The logic here is not obvious and, since there is no comment for this member in EmitContext, it is hard to tell if we are doing the right thing. A clarifying comment might help.